extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java

```diff
@@ -951,4 +953,23 @@ public boolean isAllowUserInfoCache() {
     public void setAllowUserInfoCache(boolean allowUserInfoCache) {
         this.allowUserInfoCache = allowUserInfoCache;
     }
+
+    public Optional<Provider> getProvider() {
+        return provider;
+    }
+
+    public void setProvider(Provider provider) {
+        this.provider = Optional.of(provider);
+    }
+
+    /**
+     * Well known OpenId Connect provider identifier
+     */
+    @ConfigItem
+    public Optional<Provider> provider = Optional.empty();
+
+    public enum Provider {
+        GITHUB,
+        APPLE
```

Comment:
Do we plan to support the others? I suppose we could have Google and Microsoft at some point?

Reply:
Yeah, I plan to add support for them as soon as I can.

Reply:
@gsmet `GitHub` and `Apple` require some non-trivial configuration while with Google, only the address is required in simple cases  - it is a standard OIDC provider, works the same way as with Keycloak - so for ex for `bearer` tokens all what is needed is `quarkus.oidc.auth-server-url`:
```
quarkus.oidc.auth-server-url=https://accounts.google.com
```

for `web-app` it is 
```
quarkus.oidc.auth-server-url=https://accounts.google.com
quarkus.oidc.application-type=web-app
quarkus.oidc.client-id={GOOGLE_CLIENT_ID}
quarkus.oidc.credentials.secret={GOOGLE_CLIENT_SECRET}
```

Here it is only 
```
quarkus.oidc.auth-server-url=https://accounts.google.com
quarkus.oidc.application-type=web-app
```

which is common for all the cases - I suppose it makes sense to add even these 2 properties for the users not to worry about finding this address value.

Re `Microsoft` - I'd rather postpone adding it as there are a few different Microsoft providers and one of them has a non common tenant specific `quarkus.oidc.auth-server-url` so can not be generalized;

Reply:
I added the one for `Google` as well; perhaps we can add the one for `Facebook`, though we can always do it incrementally.
@FroMage can you please type how the Facebook provider is configured ?